pkg/mapper/crd/apis/iamauthenticator/v1alpha1/types.go

```diff
@@ -35,9 +35,10 @@ type IAMIdentityMapping struct {
 
 // IAMIdentityMappingSpec is the spec for a IAMIdentityMapping resource
 type IAMIdentityMappingSpec struct {
-	ARN      string   `json:"arn"`
-	Username string   `json:"username"`
-	Groups   []string `json:"groups"`
+	ARN         string   `json:"arn"`
+	Username    string   `json:"username"`
+	ARNPatterns string   `json:"arnPatterns"`
```

Comment:
AFAICT, this doesn't belong here. The `IAMIdentityMappingSpec` refers to the relationships between a specific Kubernetes user and a specific IAM principal.

Reply:
I think you might be right. I might just roll back my changes to that feature - perhaps I'll leave that for my next PR if this makes it in :)